mozdownload/scraper.py

```diff
@@ -611,19 +618,38 @@ def build_filename(self, binary):
                            'STUB': '-stub' if self.is_stub_installer else '',
                            'EXT': self.extension}
 
+    def get_build_info(self):
+        self.version = self.query_versions(self.version)[0]
+
+    def query_versions(self, version=None):
+        """Check specified version and resolve special values"""
+        if version not in RELEASE_AND_CANDIDATE_LATEST_VERSIONS:
+            return [version]
+
+        url = urljoin(self.base_url, 'releases/')
+        parser = self._create_directory_parser(url)
+        if version:
+            versions = parser.filter(RELEASE_AND_CANDIDATE_LATEST_VERSIONS[version])
+            from distutils.version import LooseVersion
+            versions.sort(key=LooseVersion)
+            return [versions[-1]]
+        else:
+            return parser.entries
+
 
 class ReleaseCandidateScraper(ReleaseScraper):
     """Class to download a release candidate build from the Mozilla server"""
 
-    def __init__(self, version, build_number=None, *args, **kwargs):
-        self.version = version
+    def __init__(self, build_number=None, *args, **kwargs):
         self.build_number = build_number
 
```

Comment:
So, the __init__ of this used to take a non-optional version parameter. The ReleaseScraper still does. The other scrapers do not and never did. I may be misunderstanding Python here, but wouldn't the factory creating raise for ReleaseScraper if version was not there, and raise for all of the others if version is there? This seems to be an asymetric API, and since there are no direct calls to these __init__ methods, shouldn't there be a bug here, and shouldn't it have always been there?